### ISSUE TITLE
ADL_ERR_INVALID_ADL_IDX

### ISSUE THREAD

**Issue by Just-Ch33s3** [LibreHardwareMonitor/LibreHardwareMonitor#667]
Hello again, thanks alot for your tip regarding the pre versions/artifacts !
I'm able to read out more values again;
-> Part of a full report on 08.10-pre85 | WinPro10x64_21H2 ;
```
+- AMD FirePro V4900 (FireGL V) (/gpu-amd/0)
|  +- D3D            :        0        0        0 (/gpu-amd/0/load/2)
|  +- D3D 3D         :        0        0        0 (/gpu-amd/0/load/3)
|  +- D3D Copy       :        0        0        0 (/gpu-amd/0/load/4)
|  +- D3D SAMU       :        0        0        0 (/gpu-amd/0/load/5)
|  +- D3D Video Decode :        0        0        0 (/gpu-amd/0/load/6)
|  +- D3D Dedicated Memory Used :  137.219  137.219  137.219 (/gpu-amd/0/smalldata/0)
|  +- D3D Shared Memory Used :  35.1289  35.1289  35.1289 (/gpu-amd/0/smalldata/1)
```
however, getting fan speeds and temp readings still results in an error code as shown below;
```
AMD GPU

AdapterIndex: 0

Overdrive Caps

 Status: ADL_ERR_INVALID_ADL_IDX
 Supported: 0
 Enabled: 0
 Version: 0

Overdrive5 Parameters

 Status: ADL_ERR_INVALID_ADL_IDX
 NumberOfPerformanceLevels: 0
 ActivityReportingSupported: 0
 DiscretePerformanceLevels: 0
 EngineClock.Min: 0
 EngineClock.Max: 0
 EngineClock.Step: 0
 MemoryClock.Min: 0
 MemoryClock.Max: 0
 MemoryClock.Step: 0
 Vddc.Min: 0
 Vddc.Max: 0
 Vddc.Step: 0

Overdrive5 Temperature

 Status: ADL_ERR_INVALID_ADL_IDX
 Value: 0

Overdrive5 FanSpeed

 Status RPM: ADL_ERR_INVALID_ADL_IDX
 Value RPM: 0
 Status Percent: ADL_ERR_INVALID_ADL_IDX
 Value Percent: 0

Overdrive5 CurrentActivity

 Status: ADL_ERR_INVALID_ADL_IDX
 EngineClock: 0
 MemoryClock: 0
 Vddc: 0
 ActivityPercent: 0
 CurrentPerformanceLevel: 0
 CurrentBusSpeed: 0
 CurrentBusLanes: 0
 MaximumBusLanes: 0

D3D

 Id: \\?\PCI#VEN_1002&DEV_6749&SUBSYS_2B061002&REV_00#4&180f020b&0&0010#{1ca05180-a699-450a-9a0c-de4fbe3ddd89}
```
a quick research on the status code shows it's an 'invalid adapter index'
is that something you'd be able to troubleshoot or might this be some wrong configuration on my end?

Greetings, have a good day!
Chris

**Comment by Just-Ch33s3** [comment]
additional suplement/update:
by testing several old and new drivers under 08.10-pre85 i was able to receive ADL_OK status on some values like FAN and TEMPERATURE readouts again, but only if do a full report;
```
AMD GPU

AdapterIndex: 0

Overdrive Caps

 Status: ADL_ERR_NOT_SUPPORTED
 Supported: 0
 Enabled: 0
 Version: 5

Overdrive5 Parameters

 Status: ADL_OK
 NumberOfPerformanceLevels: 3
 ActivityReportingSupported: 1
 DiscretePerformanceLevels: 1
 EngineClock.Min: 5000
 EngineClock.Max: 80000
 EngineClock.Step: 500
 MemoryClock.Min: 7500
 MemoryClock.Max: 100000
 MemoryClock.Step: 500
 Vddc.Min: 0
 Vddc.Max: 1100
 Vddc.Step: 0

Overdrive5 Temperature

 Status: ADL_OK
 Value: 33

Overdrive5 FanSpeed

 Status RPM: ADL_ERR_NOT_SUPPORTED
 Value RPM: 0
 Status Percent: ADL_OK
 Value Percent: 39

Overdrive5 CurrentActivity

 Status: ADL_OK
 EngineClock: 100
 MemoryClock: 150
 Vddc: 0,9
 ActivityPercent: 0
 CurrentPerformanceLevel: 0
 CurrentBusSpeed: 2500
 CurrentBusLanes: 16
 MaximumBusLanes: 16
```
 if i do a specific sensor readout (foreach $sensor in $hw) i only receive the following;
```
Ident:'/gpu-amd/0/smalldata/0' Value:'121.1563' 
Ident:'/gpu-amd/0/smalldata/1' Value:'33.86328' 
Ident:'/gpu-amd/0/load/3' Value:'1.056838' 
Ident:'/gpu-amd/0/load/4' Value:'0' 
Ident:'/gpu-amd/0/load/2' Value:'0' 
Ident:'/gpu-amd/0/load/6' Value:'0' 
Ident:'/gpu-amd/0/load/5' Value:'0'
```
TLDR: it looks like the FAN/TEMP values of this GPU won't be collected on specific sensor readout, only on full report

**Comment by Just-Ch33s3** [comment]
marked as 'closed'
specific problem got solved by installind AMD driver 15.201.2401.1010
